src/plugins/event-plugins/__tests__/FetchPlugin.test.ts

```diff
@@ -969,4 +966,58 @@ describe('FetchPlugin tests', () => {
             segment_id: existingSegmentId
         });
     });
+
+    test('when urlsToTrace is empty then the plugin does not record a trace', async () => {
+        // Init
+        const config: PartialHttpPluginConfig = {
+            urlsToTrace: []
+        };
+
+        const plugin: FetchPlugin = new FetchPlugin(config);
+        plugin.load(xRayOnContext);
+
+        // Run
+        await fetch(URL);
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(0);
+    });
+
+    test('when the url does not match urlsToTrace then the plugin does not record a trace', async () => {
```

Comment:
question: what about the test case 'when url matches some but not all regex in urlsToTrace'?